packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CodeModelPlugin.cs

```diff
@@ -47,6 +47,22 @@ public CodeModelPlugin(GeneratorContext context)
         /// </summary>
         /// <param name="provider">The model type provider.</param>
         /// <param name="inputModel">The input model.</param>
-        public abstract IReadOnlyList<TypeProvider> GetSerializationTypeProviders(ModelProvider provider, InputModelType inputModel);
+        public virtual IReadOnlyList<TypeProvider> GetSerializationTypeProviders(TypeProvider provider, InputType inputModel)
+        {
+            try
+            {
+                if (inputModel is InputEnumType inputEnumType)
+                {
+                    if (!inputEnumType.IsExtensible)
+                    {
+                        return new List<TypeProvider> { new FixedEnumSerializationProvider(EnumProvider.Create(inputEnumType)) };
+                    }
+                }
+            }
+            catch (InvalidCastException)
+            {
+            }
+            return Array.Empty<TypeProvider>();
```

Comment:
what is the purpose of this method?
If I quote here: https://github.com/microsoft/typespec/pull/3621/files#r1647820847, if they want to have a completely different implementation of enum (fixed enum), they should start over from `TypeProvider`, or at least from `EnumProvider`, and last time when I am implemnetation enums, there is a statement that the conversion from the enum to string and vice versa is not considered as "serialization" because this is not something that usually could be changed for example when the library supports or do not support MRW.

Reply:
This method allows someone to override what serialization type providers get used for their types both enum and model.

For model we default to no serialization, and MGC.ClientModel will override to add in MrwSerialization for models and leave enums as default.